src/core/connection.c

```diff
@@ -197,6 +197,10 @@ QuicConnAlloc(
             Connection,
             CASTED_CLOG_BYTEARRAY(sizeof(Path->Route.RemoteAddress), &Path->Route.RemoteAddress));
 
+        if (QUIC_FAILED(QuicPathResolveRoute(Path))) {
+            goto Error;
+        }
```

Comment:
All this logic of making sure to resolve the route at the right time is complicated and error prone (maybe we missed a spot; maybe one these is too early). @maolson-msft  what do you think of adding support for **in** the send call? We put a flag in the `CXPLAT_ROUTE` called something like `IsResolved`, and in the send, if it's not resolved it JIT does it? We'd still need to expose this for the client first packet scenario I think (maybe not?).

Reply:
Remember, we want to potentially re-resolve the route later on when there is retransmission, so we still need to expose the API.

Do you think you missed a spot?
-new client connection
-new server connection
-path change

What else could create a new path?